components/sync15/src/request.rs

```diff
@@ -1323,14 +1324,28 @@ mod test {
         );
     }
 
+    macro_rules! assert_feq {
```

Comment:
Can we implement this comparison directly on `ServerTimestamp`? Looks like it came in handy in `bso_record.rs`, too...

Reply:
Honestly we want both of these to be direct comparisons, but eh. I actually don't like this clippy lint, and considered turning it off, but there's so little math in this repo that it's not worth it.

std::f64::EPSILON is actually wrong to use here if we wanted a relative comparison, so I don't think it should be shared. f64::EPSILON is the distance between 1.0 and the next float above 1.0. We expect the timestamps to be much larger than this, and floats get further apart as they get larger, so this is just not right.

If we actually wanted to compare for floats for approximate equality, we'd probably want something like `(t0 - t1).abs() <= THRESHOLD * t0.abs().max(t1.abs()).max(1.0)`, where THRESHOLD is something we would have to define (see http://realtimecollisiondetection.net/blog/?p=89 for some discussion on how you might derive this, but note that EPSILON there is not the same as std::f64::EPSILON).

However, in practice, we know ServerTimestamps are seconds with resolution to the hundredths place, and so *really* we should just convert to an integer to avoid such issues.

I'll file a followup for us to make ServerTimestamp store milliseconds instead of being a f64.

Reply:
Filed #815